internal/command/meta_backend.go

```diff
@@ -1649,6 +1665,135 @@ func (m *Meta) backendInitFromConfig(c *configs.Backend) (backend.Backend, cty.V
 	return b, configVal, diags
 }
 
+// stateStoreInitFromConfig returns an initialized and configured state store, using the backend.Backend interface.
+// During this process:
+// > The provider is configured, after validating provider config
+// > The state store is configured, after validating state_store config
+//
+// NOTE: the backend version of this method, `backendInitFromConfig`, prompts users for input if any required fields
+// are missing from the backend config. In `stateStoreInitFromConfig` we don't do this, and instead users will see an error.
+func (m *Meta) stateStoreInitFromConfig(c *configs.StateStore, factory providers.Factory) (backend.Backend, cty.Value, cty.Value, tfdiags.Diagnostics) {
```

Comment:
Highlighting this - I this prompting users for input when required fields aren't set is an odd feature and I'm not sure if it's used much. Is this to help users who forget to set values via config or their environment, or is this a first-class feature that's used to pass values into Terraform?

I'm assuming it's a legacy-ish feature that we wouldn't choose to add now, so figured we exclude it to start and we can wait for feature requests in future.